### PR TITLE
Profile operations

### PR DESCRIPTION
**Pull request by shreyasthiagaraj**
This PR adds support for the various profile-related methods, data types, and helper functions in Blockstack.js. Specific methods:

- `extractProfile`
- `wrapProfileToken`
- `signProfileToken`
- `verifyProfileToken`
- `lookupProfile`

All functions also have Objective-C bindings, found in `Blockstack+ObjC.swift`.

Thanks!
Shreyas

### CODE REVIEW

File: Tools/Blockstack-webapp/public/redirect.html [hunk 1]
```diff
@@ -1,15 +1,15 @@
...
+  <title>Afari Redirect</title>
   <script>
   function getParameterByName(name) {
     var match = RegExp('[?&]' + name + '=([^&]*)').exec(window.location.search);
     return match && decodeURIComponent(match[1].replace(/\+/g, ' '));
   }
 
   var authResponse = getParameterByName('authResponse')
-  window.location="myblockstackapp://?authResponse=" + authResponse
+  window.location="afari://?authResponse=" + authResponse
```
Comment (larrysalibra) [inline Tools/Blockstack-webapp/public/redirect.html]: is this supposed to be here?

**Review by larrysalibra — CHANGES_REQUESTED** [review]
Generally looks good to me.

Couple of things:

* We shouldn't commit the  `.DS_Store` files to git.
* I'd like to see test coverage on the newly exposed profile methods. There's a test for one method, but don't see anything for the others.

**Comment by shreyasthiagaraj** [comment]
@larrysalibra .DS_Store files are being ignored now, and all profile methods now have tests!

File: Example/Tests/ProfileTests.swift [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJqdGkiOiI2OGExZjZiMy1hNmZlLTQ1ZWEtYTY0OC0xM2JmZDQ1NjMwYjQiLCJpYXQiOiIyMDE4LTEwLTIyVDEyOjM3OjIyLjg5NFoiLCJleHAiOiIyMDE5LTEwLTIyVDEyOjM3OjIyLjg5NFoiLCJzdWJqZWN0Ijp7InB1YmxpY0tleSI6IjAzNjRmYmE0ZDIzZDY0ZTNhMWUxZDJmYjQwZDU4YjMwOWExMTMxYWNiMThiZWNhMjFhNjkzYzliYmQ2YjZjMzE0ZSJ9LCJpc3N1ZXIiOnsicHVibGljS2V5IjoiMDM2NGZiYTRkMjNkNjRlM2ExZTFkMmZiNDBkNThiMzA5YTExMzFhY2IxOGJlY2EyMWE2OTNjOWJiZDZiNmMzMTRlIn0sImNsYWltIjp7IkB0eXBlIjoiUGVyc29uIiwiQGNvbnRleHQiOiJodHRwOi8vc2NoZW1hLm9yZyIsIm5hbWUiOiJUZXN0aW5nMTIzIiwiZGVzY3JpcHRpb24iOiJ0ZXN0ZXIgdGVzdGluZyB0ZXN0cyJ9fQ.BEvMyIU72J8p4xE2Yrcqnq4ZO-DfsPV-jqwCv0-dfYpztr1VtAKrF7MFSS4sMRrSfHdTUEAE1VHe9HLulwBOmA",
+                "decodedToken": {
+                  "header": {
+                    "typ": "JWT",
+                    "alg": "ES256K"
+                  },
+                  "payload": {
+                    "jti": "68a1f6b3-a6fe-45ea-a648-13bfd45630b4",
+                    "iat": "2018-10-22T12:37:22.894Z",
+                    "exp": "2019-10-22T12:37:22.894Z",
```
Comment (larrysalibra) [inline Example/Tests/ProfileTests.swift:31]: These tests will start failing in a year - might want to make a note in a comment.

**Review by larrysalibra — APPROVED** [review]
Thanks for the changes! I made a couple small nitpicky comments, but besides that, looks good to go!

**Pull request merged**